### PR TITLE
Add DeviceWrappingPlug.

### PR DESCRIPTION
**Pull request by jettisonjoe**

### CODE REVIEW

File: openhtf/plugs/device_wrapping.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+    if not callable(attribute):
+      return attribute
+
+    def arg_string(arg):
+      """Returns a stdout-friendly string representation of the argument"""
+      arg_repr = repr(arg)
+      return arg_repr if len(arg_repr) < 40 else '<{} of length {}>'.format(
+          type(arg).__name__, len(arg_repr))
+
```
Comment (kdsudac) [inline openhtf/plugs/device_wrapping.py]: Comments here to explain.

# Attribute it callable, so return wrapped method that logs calls with args & kwargs.

File: openhtf/plugs/device_wrapping.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+  """A base plug for wrapping existing device abstractions.
+
+  Subclass instances must override the _device attribute to which normal
+  attribute access will be delegated. Subclasses can use the
+  @conf.inject_positional_args decorator on their constructors to get any
+  configuration needed to construct the inner device instance.
+
+  Example:
+    class BleSnifferPlug(DeviceWrappingPlug):
+      ...
```
[the remark this answers is not in the document]
Comment (jettisonjoe) [reply openhtf/plugs/device_wrapping.py]: Oops, that docstring was a little misleading / out-of-date. I reworded it to be more clear.

File: openhtf/plugs/device_wrapping.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    if self._device is None:
+      raise openhtf.plugs.InvalidPlugError(
+        'DeviceWrappingPlug instances must set the _device attribute.')
+
+    attribute = getattr(self._device, attr)
+
+    if not callable(attribute):
+      return attribute
+
+    def arg_string(arg):
```
[the remark this answers is not in the document]
Comment (jettisonjoe) [reply openhtf/plugs/device_wrapping.py]: Good point. Moved it out to the module namespace and renamed for clarity.

File: openhtf/plugs/device_wrapping.py (hunk 1 above)
Comment (jettisonjoe) [reply openhtf/plugs/device_wrapping.py]: Done.

**Pull request merged**